libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -122,31 +246,46 @@ export function computedAsync<T>(
 			options?.behavior ?? 'switch',
 		);
 
-		const sourceResult = source$.subscribe({
-			next: (value) => sourceValue.set(value),
-			error: (error) => {
+		const sourceResult = source$
+			// we skip the first value if requireSync is true because we already set the value in the sourceValue
+			.pipe(options?.requireSync ? skip(1) : (x) => x)
+			.subscribe({
+				next: (value) => sourceValue.set({ kind: StateKind.Value, value }),
 				// NOTE: Error should be handled by the user (using catchError or .catch())
-				sourceValue.set(error);
-			},
-		});
+				error: (error) => sourceValue.set({ kind: StateKind.Error, error }),
+			});
 
 		destroyRef.onDestroy(() => {
 			effectRef.destroy();
 			sourceResult.unsubscribe();
 		});
 
+		if (options?.requireSync && sourceValue()!.kind === StateKind.NoValue) {
```

Comment:
```suggestion
		if (options?.requireSync && sourceValue().kind === StateKind.NoValue) {
```
Looks like the assertion isn't required